src/cpp/database/database.hpp

```diff
@@ -212,22 +212,29 @@ class Database
     /**
      * Get all entities of a given EntityKind related to another entity
      *
+     * In case the \c entity_id is EntityId::all(), all entities of type \c entity_type are returned
+     *
      * @param entity_id constant reference to the EntityId of the entity to which the returned
      *                  entities are related
      * @param entity_kind The EntityKind of the fetched entities
-     * @throws eprosima::statistics_backend::BadParameter if there is no entity with the given ID.
+     * @throws eprosima::statistics_backend::BadParameter if there is no entity with the given ID and
+     * is not Entity::all().
```

Comment:
```suggestion
     * @throws eprosima::statistics_backend::BadParameter if there is no entity with the given ID or
     * if entity_id is not EntityId::all().
```

It also throws when `entity_kind = EntityKind::INVALID`. Please, complete documentation and take into account this suggestion for other similar comments.